From d16b8d44cd5cb85fb0f0d0ff0fbb60d8d90c6646 Mon Sep 17 00:00:00 2001
From: Sergey Korolev <s.korolev@ndmsystems.com>
Date: Tue, 12 Jun 2018 22:07:02 +0300
Subject: [PATCH] prevent native gdb "unknown type name 'gdb_fpregset_t'"
 errors

when building with glibc

../gdb/gdb_proc_service.h:173:9: error: unknown type name 'gdb_fpregset_t'
../gdb/gdb/gdb_proc_service.h:173:9: error: unknown type name 'gdb_fpregset_t'
../gdb/gdb/proc-service.c:197:1: error: conflicting types for 'ps_lgetfpregs'
../gdb/gdb/proc-service.c:217:1: error: conflicting types for 'ps_lsetfpregs'
../gdb/gdb/gdb_proc_service.h:173:9: error: unknown type name 'gdb_fpregset_t'
---
 gdb/gdb_proc_service.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/gdb/gdb_proc_service.h b/gdb/gdb_proc_service.h
index 04d3c03..54c4195 100644
--- a/gdb/gdb_proc_service.h
+++ b/gdb/gdb_proc_service.h
@@ -20,6 +20,7 @@
 #define GDB_PROC_SERVICE_H
 
 #include <sys/types.h>
+#include "gregset.h"
 
 #ifdef HAVE_PROC_SERVICE_H
 
@@ -60,8 +61,6 @@ EXTERN_C_POP
 #include <sys/procfs.h>
 #endif
 
-#include "gregset.h"
-
 EXTERN_C_PUSH
 
 /* Functions in this interface return one of these status codes.  */
-- 
2.11.0

